src/Util/FilterUtil.ts

```diff
@@ -71,6 +80,182 @@ class FilterUtil {
   //   return filter;
   // }
 
+  /**
+   * Checks if a feature matches the specified filter.
+   * Returns true if it matches, otherwise returns false.
+   */
+  static featureMatchesFilter = (filter: Filter, feature: any): boolean => {
+    if (filter.length === 0) {
+      return true;
+    }
+    let matchesFilter: boolean = true;
+    const operator: Operator = filter[0];
+    let isNestedFilter: boolean = false;
+    if (FilterUtil.operatorMapping[operator]) {
+      isNestedFilter = true;
+    }
+    if (isNestedFilter) {
+      switch (filter[0]) {
+        case '&&':
+          let intermediate = true;
+          let restFilter = filter.slice(1);
+          restFilter.forEach((f: Filter) => {
+            if (!FilterUtil.featureMatchesFilter(f, feature)) {
+              intermediate = false;
+            }
+          });
+          matchesFilter = intermediate;
+          break;
+        case '||':
+          intermediate = false;
+          restFilter = filter.slice(1);
+          restFilter.forEach((f: Filter) => {
+            if (FilterUtil.featureMatchesFilter(f, feature)) {
+              intermediate = true;
+            }
+          });
+          matchesFilter = intermediate;
+          break;
+        case '!':
+          matchesFilter = !FilterUtil.featureMatchesFilter(filter[1], feature);
+          break;
+        default:
+          throw new Error(`Cannot parse Filter. Unknown combination or negation operator.`);
+      }
+    } else {
+      const prop: any = _get(feature, 'properties[' + filter[1] + ']');
+      switch (filter[0]) {
+        case '==':
+          // tslint:disable-next-line
```

Comment:
I used the equality operator (`==`) instead of the identity operator (`===`) there. Got a little confused after yesterday's discussion about parsing properties in sld.
Solved in https://github.com/terrestris/geostyler/pull/639/commits/1300ace7450b8804767bd6964d5ad38a18f83e9f